astropy/io/fits/tests/test_fitsheader.py

```diff
@@ -0,0 +1,159 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import sys
+
+import numpy as np
+import re
+
+from . import FitsTestCase
+from ..hdu import PrimaryHDU
+from ..scripts import fitsheader
+from ....tests.helper import catch_warnings
+from ....tests.helper import pytest
+from ....utils.exceptions import AstropyDeprecationWarning
+from ....version import version
+ 
+# @pytest.fixture(scope=module , params=['./tests/test0.fits'])
+# def resource_testfile_setup(request):
+#     with open(request.params[0]) as test_fits:
+#         yield test_fits
+
+@pytest.fixture(scope='module')
+def warning_regex():
+    regex = re.compile('^WARNING.*')
+    return regex
+
+@pytest.fixture(scope='module')
+def error_regex():
+    regex = re.compile('^ERROR.*')
+    return regex
+
+class TestFITSheader_script(FitsTestCase):
+
+    def setup_method(self,  method):
+        self.sys_argv_orig = sys.argv
+        sys.argv = ["fitsheader"]
+
+    def teardown_method(self, method):
+        sys.argv = self.sys_argv_orig
+
+    def test_noargs(self):
+        with pytest.raises(SystemExit) as e:
+            fitsheader.main()
+        assert e.value.code == 2
+
+    @pytest.mark.parametrize('test_file', [
+        ('test0.fits'),
+        pytest.mark.xfail(('random.fits'), reason = 'bad file')
```

Comment:
- `xfail` - should work but currently doesn't because of some unfixed bug (either in-package or upstream)
- `skip` - doesn't work if dependency is too old or if dependency is not installed
- `pytest.raises` - should fail, make sure it fails with the right exception (and exception message).

That's not official, that's just how I interpret these.